mode of mkdir incorrect after setacl.

**What happened**:
```
root@bench-01:~/juicefs# cat .github/scripts/hypo/acl_test.py
    def test_acl_5000(self):
        state = JuicefsMachine()
        v1 = state.init_folders()
        v3 = state.set_acl(default=True, entry=v1, group=v1, group_perm=set(), logical=False, mask=set(), not_recalc_mask=False, other_perm=set(), physical=False, recalc_mask=False, recursive=False, set_mask=False, sudo_user='root', user=v1, user_perm={'r'})
        state.mkdir(mode=0, parent=v1, subdir='aaaa', umask=0, user='root')
        state.teardown()

root@bench-01:~/juicefs# python3 .github/scripts/hypo/acl_test.py -k test_acl_5000
setup_logger ./log1
setup_logger ./log2
__init__
duration is 0.026491165161132812
2024-04-08 17:41:30,708 - INFO - run_cmd: sudo -u root setfacl -d      -m u::r,g::-,o::- /tmp/fsrand/
run_cmd:getfacl /tmp/fsrand/
2024-04-08 17:41:30,714 - INFO - do_set_acl /tmp/fsrand/ with u::r,g::-,o::- succeed
2024-04-08 17:41:30,714 - INFO - run_cmd: sudo -u root setfacl -d      -m u::r,g::-,o::- /tmp/jfs/fsrand/
run_cmd:getfacl /tmp/jfs/fsrand/
2024-04-08 17:41:30,720 - INFO - do_set_acl /tmp/jfs/fsrand/ with u::r,g::-,o::- succeed
2024-04-08 17:41:30,720 - INFO - do_mkdir /tmp/fsrand/aaaa with mode 0o0 succeed
2024-04-08 17:41:30,720 - INFO - do_mkdir /tmp/jfs/fsrand/aaaa with mode 0o0 succeed
F
======================================================================
FAIL: test_acl_5000 (__main__.TestFsrand2)
----------------------------------------------------------------------
Traceback (most recent call last):
  File ".github/scripts/hypo/acl_test.py", line 123, in test_acl_5000
    state.mkdir(mode=0, parent=v1, subdir='aaaa', umask=0, user='root')
  File "/root/juicefs/.github/scripts/hypo/fsrand2.py", line 315, in mkdir
    parent = Folders.filter(lambda x: x != multiple()),
  File "/root/hypothesis/hypothesis-python/src/hypothesis/stateful.py", line 681, in rule_wrapper
    return f(*args, **kwargs)
  File "/root/juicefs/.github/scripts/hypo/fsrand2.py", line 315, in mkdir
    parent = Folders.filter(lambda x: x != multiple()),
  File "/root/hypothesis/hypothesis-python/src/hypothesis/stateful.py", line 802, in precondition_wrapper
    return f(*args, **kwargs)
  File "/root/juicefs/.github/scripts/hypo/fsrand2.py", line 326, in mkdir
    assert self.equal(result1, result2), f'\033[31mmkdir:\nresult1 is {result1}\nresult2 is {result2}\033[0m'
AssertionError: mkdir:
result1 is (0, 0, '0o40000')
result2 is (0, 0, '0o40400')

----------------------------------------------------------------------
Ran 1 test in 0.028s

FAILED (failures=1)
```


**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**

**Environment**:
- JuiceFS version (use `juicefs --version`) or Hadoop Java SDK version:
- Cloud provider or hardware configuration running JuiceFS:
- OS (e.g `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Object storage (cloud provider and region, or self maintained):
- Metadata engine info (version, cloud provider managed or self maintained):
- Network connectivity (JuiceFS to metadata engine, JuiceFS to object storage):
- Others:

--- a/pkg/meta/base_test.go
+++ b/pkg/meta/base_test.go
@@ -122,37 +122,37 @@ func testMeta(t *testing.T, m Meta) {
 	}
 
 	testMetaClient(t, m)
-	testTruncateAndDelete(t, m)
-	testTrash(t, m)
-	testParents(t, m)
-	testRemove(t, m)
-	testResolve(t, m)
-	testStickyBit(t, m)
-	testLocks(t, m)
-	testListLocks(t, m)
-	testConcurrentWrite(t, m)
-	testCompaction(t, m, false)
-	time.Sleep(time.Second)
-	testCompaction(t, m, true)
-	testCopyFileRange(t, m)
-	testCloseSession(t, m)
-	testConcurrentDir(t, m)
-	testAttrFlags(t, m)
-	testQuota(t, m)
-	testAtime(t, m)
-	testAccess(t, m)
-	base := m.getBase()
-	base.conf.OpenCache = time.Second
-	base.of.expire = time.Second
-	testOpenCache(t, m)
-	base.conf.CaseInsensi = true
-	testCaseIncensi(t, m)
-	testCheckAndRepair(t, m)
-	testDirStat(t, m)
-	testClone(t, m)
+	// testTruncateAndDelete(t, m)
+	// testTrash(t, m)
+	// testParents(t, m)
+	// testRemove(t, m)
+	// testResolve(t, m)
+	// testStickyBit(t, m)
+	// testLocks(t, m)
+	// testListLocks(t, m)
+	// testConcurrentWrite(t, m)
+	// testCompaction(t, m, false)
+	// time.Sleep(time.Second)
+	// testCompaction(t, m, true)
+	// testCopyFileRange(t, m)
+	// testCloseSession(t, m)
+	// testConcurrentDir(t, m)
+	// testAttrFlags(t, m)
+	// testQuota(t, m)
+	// testAtime(t, m)
+	// testAccess(t, m)
+	// base := m.getBase()
+	// base.conf.OpenCache = time.Second
+	// base.of.expire = time.Second
+	// testOpenCache(t, m)
+	// base.conf.CaseInsensi = true
+	// testCaseIncensi(t, m)
+	// testCheckAndRepair(t, m)
+	// testDirStat(t, m)
+	// testClone(t, m)
 	testACL(t, m)
-	base.conf.ReadOnly = true
-	testReadOnly(t, m)
+	// base.conf.ReadOnly = true
+	// testReadOnly(t, m)
 }
 
 func testAccess(t *testing.T, m Meta) {
@@ -373,6 +373,7 @@ func testACL(t *testing.T, m Meta) {
 		t.Fatalf("create %s: %s", subDir, st)
 	}
 	defer m.Rmdir(ctx, testDirIno, subDir2)
+	assert.Equal(t, uint16(0), attr2.Mode)
 
 	// subdir inherit default acl
 	rule3 = &aclAPI.Rule{}
@@ -385,12 +386,6 @@ func testACL(t *testing.T, m Meta) {
 	rule3 = &aclAPI.Rule{}
 	st = m.GetFacl(ctx, subDirIno2, aclAPI.TypeAccess, rule3)
 	assert.Equal(t, ENOATTR, st)
-
-	attr2 = &Attr{}
-	if st := m.GetAttr(ctx, subDirIno2, attr2); st != 0 {
-		t.Fatalf("getattr error: %s", st)
-	}
-	assert.Equal(t, rule.GetMode(), attr2.Mode)
 
 	// test cache all
 	sz := m.getBase().aclCache.Size()

--- a/pkg/meta/base_test.go
+++ b/pkg/meta/base_test.go
@@ -122,37 +122,37 @@ func testMeta(t *testing.T, m Meta) {
 	}
 
 	testMetaClient(t, m)
-	// testTruncateAndDelete(t, m)
-	// testTrash(t, m)
-	// testParents(t, m)
-	// testRemove(t, m)
-	// testResolve(t, m)
-	// testStickyBit(t, m)
-	// testLocks(t, m)
-	// testListLocks(t, m)
-	// testConcurrentWrite(t, m)
-	// testCompaction(t, m, false)
-	// time.Sleep(time.Second)
-	// testCompaction(t, m, true)
-	// testCopyFileRange(t, m)
-	// testCloseSession(t, m)
-	// testConcurrentDir(t, m)
-	// testAttrFlags(t, m)
-	// testQuota(t, m)
-	// testAtime(t, m)
-	// testAccess(t, m)
-	// base := m.getBase()
-	// base.conf.OpenCache = time.Second
-	// base.of.expire = time.Second
-	// testOpenCache(t, m)
-	// base.conf.CaseInsensi = true
-	// testCaseIncensi(t, m)
-	// testCheckAndRepair(t, m)
-	// testDirStat(t, m)
-	// testClone(t, m)
+	testTruncateAndDelete(t, m)
+	testTrash(t, m)
+	testParents(t, m)
+	testRemove(t, m)
+	testResolve(t, m)
+	testStickyBit(t, m)
+	testLocks(t, m)
+	testListLocks(t, m)
+	testConcurrentWrite(t, m)
+	testCompaction(t, m, false)
+	time.Sleep(time.Second)
+	testCompaction(t, m, true)
+	testCopyFileRange(t, m)
+	testCloseSession(t, m)
+	testConcurrentDir(t, m)
+	testAttrFlags(t, m)
+	testQuota(t, m)
+	testAtime(t, m)
+	testAccess(t, m)
+	base := m.getBase()
+	base.conf.OpenCache = time.Second
+	base.of.expire = time.Second
+	testOpenCache(t, m)
+	base.conf.CaseInsensi = true
+	testCaseIncensi(t, m)
+	testCheckAndRepair(t, m)
+	testDirStat(t, m)
+	testClone(t, m)
 	testACL(t, m)
-	// base.conf.ReadOnly = true
-	// testReadOnly(t, m)
+	base.conf.ReadOnly = true
+	testReadOnly(t, m)
 }
 
 func testAccess(t *testing.T, m Meta) {

--- a/pkg/meta/random_test.go
+++ b/pkg/meta/random_test.go
@@ -219,7 +219,7 @@ func (m *fsMachine) create(_type uint8, parent Ino, name string, mode, umask uin
 
 		if rule.IsMinimal() {
 			// simple acl as default
-			n.mode = (mode & 0xFE00) | rule.GetMode()
+			n.mode = mode & (0xFE00 | rule.GetMode())
 		} else {
 			cRule := rule.ChildAccessACL(mode)
 			n.accACL = cRule

--- a/pkg/meta/redis.go
+++ b/pkg/meta/redis.go
@@ -1255,7 +1255,7 @@ func (m *redisMeta) doMknod(ctx Context, parent Ino, name string, _type uint8, m
 
 			if rule.IsMinimal() {
 				// simple acl as default
-				attr.Mode = (mode & 0xFE00) | rule.GetMode()
+				attr.Mode = mode & (0xFE00 | rule.GetMode())
 			} else {
 				cRule := rule.ChildAccessACL(mode)
 				id, err := m.insertACL(ctx, tx, cRule)

--- a/pkg/meta/sql.go
+++ b/pkg/meta/sql.go
@@ -1335,7 +1335,7 @@ func (m *dbMeta) doMknod(ctx Context, parent Ino, name string, _type uint8, mode
 
 			if rule.IsMinimal() {
 				// simple acl as default
-				n.Mode = (mode & 0xFE00) | rule.GetMode()
+				n.Mode = mode & (0xFE00 | rule.GetMode())
 			} else {
 				cRule := rule.ChildAccessACL(mode)
 				id, err := m.insertACL(s, cRule)

--- a/pkg/meta/tkv.go
+++ b/pkg/meta/tkv.go
@@ -1147,7 +1147,7 @@ func (m *kvMeta) doMknod(ctx Context, parent Ino, name string, _type uint8, mode
 
 			if rule.IsMinimal() {
 				// simple acl as default
-				attr.Mode = (mode & 0xFE00) | rule.GetMode()
+				attr.Mode = mode & (0xFE00 | rule.GetMode())
 			} else {
 				cRule := rule.ChildAccessACL(mode)
 				id, err := m.insertACL(tx, cRule)

